meerkat_frontend/src/js/technical/tabletSubmission.js

```diff
@@ -0,0 +1,320 @@
+                                                    addPaginationListener('#clinicsTable table');
+                                                    colorClinicsRow();
+                                                });
+
+                                            });
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+    });
+}
+
+
+function buildTableStructure() {
+    //Add the first columns
+    buildTableColumn("DevicesIDs", "Devices IDs", "");
+    buildTableColumn("clinicType", "clinic type", "");
+    buildTableColumn("caseType", "case type", "");
+
+    for (i = week - 2; i <= week; i++) {
+        //Draw the clolumns depend on the week number
+        buildTableColumn("casesInWeek" + i, "cases in Week " + i, "");
+    }
+
+    for (i = week - 2; i <= week; i++) {
+        //Draw the clolumns depend on the week number
+        buildTableColumn("noOfRegistersInWeek" + i, "no. of registers in week " + i, "");
+    }
+
+    buildTableColumn("casesInMonthC", "cases in month (current)", "");
+    buildTableColumn("casesInMonthL", "cases in month (before the current)", "");
+
+    for (i = current_Year - 2; i <= current_Year; i++) {
+        //Draw the clolumns depend on the year number ...
+        buildTableColumn("casesInYear" + i, "cases in year " + i, "");
+    }
+}
+
+
+//This function is responsible for building the table header
+function buildTableColumn(columnId, columnName, columnClass) {
+    columnNameArray.push({
+        field: columnId,
+        title: i18n.gettext(columnName),
+        align: "center",
+        sortable: true,
+        sortName: columnId,
+        valign: "middle",
+        class: columnClass
+    });
+}
+
+
+//Fill the table for the fisrt time and add defualt data
+function fillTabletData(clinicObj) {
+    var dataObject = {};
+
+    //Add Clinic data
+    dataObject.DevicesIDs = clinicObj.clinicId;
+    dataObject.clinicType = " - ";
+    dataObject.caseType = " - ";
+
+    for (i = week - 2; i <= week; i++) {
+        dataObject["casesInWeek" + i] = "-";
+    }
+
+    for (i = week - 2; i <= week; i++) {
+        dataObject["noOfRegistersInWeek" + i] = "-";
+    }
+
+    dataObject.casesInMonthC = "-";
+    dataObject.casesInMonthL = "-";
+
+    for (i = current_Year - 2; i <= current_Year; i++) {
+        //Draw the clolumns depend on the week number ...
+        dataObject["casesInYear" + i] = "-";
+    }
+    tabletDataArray.push(dataObject);
+
+
+    //Add devices data
+    $.each(clinicObj.deviceSubmissions, function(index, value) {
+        dataObject = {};
+        dataObject.DevicesIDs = "- DeviceId  " + value.deviceId;
+        dataObject.clinicType = " - ";
+        dataObject.caseType = " - ";
+
+        for (i = week - 2; i <= week; i++) {
```

Comment:
Maybe you could just do:
```
          dataObject["casesInWeek" + (week - 2)] = value.submissionsCount;

+        for (i = week - 1; i <= week; i++) {
 +                dataObject["casesInWeek" + i] = "-";
 +        }
```
I think it's more readable.

Reply:
i think after merging the loops, things changed , where i can use (week-2) in two places